### PR TITLE
Several fixes to the CCCC reader/writers

### PR DESCRIPTION
**Pull request by ntouran**
This fixes some issues related to the recent upgrades/changes to the CCCC reader/writers

* Fixes a class inheritance propagation bug affecting the new adjoint RTFLUX reader/writer
* Propagates that fix to other related classes in case they get subclassed later
* Fixes an i/j order bug on the new GEODST reader
* Implements a rwIntMatrix method and applies use of the related public methods 
   rather than the private _rwMatrix
* Adds new tests covering these bugs

### CODE REVIEW

File: armi/nuclearDataIO/cccc.py [hunk 1]
```diff
@@ -356,6 +356,12 @@ def rwDoubleMatrix(self, contents, *shape):
         """
         return self._rwMatrix(contents, self.rwDouble, *shape)
 
+    def rwIntMatrix(self, contents, *shape):
+        """
+        Read or write a matrix of int values.
+        """
+        return self._rwMatrix(contents, self.rwInt, *shape)
+
```
Comment (youngmit) [inline armi/nuclearDataIO/cccc.py:364]: While we're lookin at them, these could both be static methods

Comment (youngmit) [reply armi/nuclearDataIO/cccc.py:364]: My guess is there's a numpy function that could take a type and a data buffer and some size/stride info and turn it into an array. If we ever care enough about speed, might be worth thinking about using that below

File: armi/nuclearDataIO/rzflux.py [hunk 2]
```diff
@@ -146,7 +146,7 @@ def _rw2DRecord(self):
         Zones are blocked into multiple records so we have to block or unblock
         them.
 
-        _rwMatrix reverses the indices into FORTRAN data order so be
+        rwMatrix reverses the indices into FORTRAN data order so be
```
Comment (youngmit) [inline armi/nuclearDataIO/rzflux.py:149]: Worth considering https://numpy.org/doc/stable/reference/generated/numpy.frombuffer.html
That would read into a 1-D, then https://numpy.org/doc/stable/reference/generated/numpy.reshape.html#numpy.reshape to reshape. It even takes an arg for column-major or row major.

File: armi/nuclearDataIO/isotxs.py [hunk 3]
```diff
@@ -175,7 +175,7 @@ def _write(cls, lib, fileName, fileMode):
 
     @classmethod
     def _readWrite(cls, lib, fileName, fileMode, getNuclideFunc):
-        with _IsotxsIO(fileName, lib, fileMode, getNuclideFunc) as rw:
+        with cls(fileName, lib, fileMode, getNuclideFunc) as rw:
```
Comment (youngmit) [inline armi/nuclearDataIO/isotxs.py:178]: Still seems kind of squirrely

**Review by youngmit — APPROVED** [review]
Just a couple of recommendations. In general, I think the way all of these classes interact is pretty confusing. But if they work it really doesnt matter as long as the public interface is small and well-documented. The structure kind of makes that difficult, though...

File: armi/nuclearDataIO/isotxs.py (hunk 3 above)
Comment (ntouran) [reply armi/nuclearDataIO/isotxs.py:178]: 🐿️  
was squirrelier before when it was demolishing any inheritance by having the class be hard-coded!

File: armi/nuclearDataIO/rzflux.py (hunk 2 above)
Comment (ntouran) [reply armi/nuclearDataIO/rzflux.py:149]: great idea. Speed does kinda matter when we're reading big scatter matrices and stuff.

**Pull request merged**